File: crazyflie/scripts/crazyflie_add.py
#!/usr/bin/env python

import rospy
from crazyflie.srv import *

if __name__ == "__main__":
    rospy.init_node('crazyflie_add', anonymous=True)
    uri = rospy.get_param("~uri")
    tf_prefix = rospy.get_param("~tf_prefix")
    roll_trim = float(rospy.get_param("~roll_trim", "0"))
    pitch_trim = float(rospy.get_param("~pitch_trim", "0"))
    enable_logging = bool(rospy.get_param("~enable_logging", "True"))
    rospy.loginfo("wait_for_service /add_crazyflie")
    rospy.wait_for_service('/add_crazyflie')
    rospy.loginfo("found /add_crazyflie")
    add_crazyflie = rospy.ServiceProxy('/add_crazyflie', AddCrazyflie)
    add_crazyflie(uri, tf_prefix, roll_trim, pitch_trim, enable_logging)

File: crazyflie/src/crazyflie_add.cpp
#include "ros/ros.h"
#include "crazyflie/AddCrazyflie.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "crazyflie_add", ros::init_options::AnonymousName);
  ros::NodeHandle n("~");

  // read paramaters
  std::string uri;
  std::string tf_prefix;
  double roll_trim;
  double pitch_trim;
  bool enable_logging;

  n.getParam("uri", uri);
  n.getParam("tf_prefix", tf_prefix);
  n.param("roll_trim", roll_trim, 0.0);
  n.param("pitch_trim", pitch_trim, 0.0);
  n.param("enable_logging", enable_logging, true);

  ROS_INFO("wait_for_service /add_crazyflie");
  ros::ServiceClient addCrazyflieService = n.serviceClient<crazyflie::AddCrazyflie>("/add_crazyflie");
  addCrazyflieService.waitForExistence();
  ROS_INFO("found /add_crazyflie");
  crazyflie::AddCrazyflie addCrazyflie;
  addCrazyflie.request.uri = uri;
  addCrazyflie.request.tf_prefix = tf_prefix;
  addCrazyflie.request.roll_trim = roll_trim;
  addCrazyflie.request.pitch_trim = pitch_trim;
  addCrazyflie.request.enable_logging = enable_logging;
  addCrazyflieService.call(addCrazyflie);

  return 0;
}
